### ISSUE TITLE
angular.json fileReplacements not longer accept json

### ISSUE THREAD

**Issue by raphael22** [angular/angular-cli#19414]
# 🐞 bug report

### Affected Package
@angular/cli or compiler ?

### Is this a regression?

Yes, updating from 10.2.0.

### Description
`ng serve` doesn't work anymore if angular.json fileReplacements override a json file.

ex: 
`{
                  "replace": "src/assets/conf/config.json",
                  "with": "src/assets/conf/config.test.json"
                }`
prompt the following error in vscode and console: 
`String does not match the pattern of "\.([cm]?j|t)sx?$".`

## 🔬 Minimal Reproduction

Couldn't find a angular/cli 11 on stackblitz.
Very simple to reproduce.

## 🔥 Exception or Error
```
Schema validation failed with the following errors:
  Data path ".fileReplacements[1]" should NOT have additional properties(replace).
  **Data path ".fileReplacements[1].replace" should match pattern "\.([cm]?j|t)sx?$".**
  Data path ".fileReplacements[1]" should match exactly one schema in oneOf.
```

## 🌍  Your Environment

**Angular Version:** 11

**Comment by alan-agius4** [comment]
Hi @raphael22, actually in your case probably it's more related to #19390 unless you are using `resolveJsonModules` and importing the json.

**Comment by raphael22** [comment]
Fixed in @angular/cli 11.0.2.
